### PR TITLE
express server including the 'get worlds', 'get layers' and 'upload files'

### PR DESCRIPTION
**Pull request by dafnaweiss**

### CODE REVIEW

File: api/uploadFileMethods.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
+const express = require('express');
+const config = require('../config/configJson');
+const fs = require('fs-extra');
+const { execSync } = require('child_process');          // for using the cURL command line
+
+const app = express();
```
Comment (yairtawil) [inline api/uploadFileMethods.js]: remove

File: api/uploadFiles.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+const formidable = require('express-formidable');
+const config = require('../config/configJson');
+const fs = require('fs-extra');
+const { execSync } = require('child_process');          // for using the cURL command line
+const path = require('path');
+const zip = require('express-easy-zip');
+// const archiver = require('archiver');
+require('./uploadFileMethods')();
+
+const app = express();
```
Comment (yairtawil) [inline api/uploadFiles.js]: use Router

File: api/uploadFiles.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
+const express = require('express');
+const router = express.Router();
```
Comment (yairtawil) [inline api/uploadFiles.js:2]: const app  =  express.Router();

File: api/uploadFiles.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+        // 5. delete all the uncompleted tasks in the import queue
+        deleteUncompleteImports();
+    }
+
+});
+
+module.exports = app;
+
+/*
```
Comment (yairtawil) [inline api/uploadFiles.js]: remove comments

File: api/worlds.js [hunk 5]
```diff
@@ -1,18 +1,19 @@
 const express = require('express');
 const router = express.Router();
 const axios = require('axios');
-const { ipAddress, serverPort, geoserverPort, Authorization } = require('../config/config');
+const config = require('../config/configJson');
 
-const urlGeoserverRest = `http://${ipAddress}${geoserverPort}/geoserver/rest/workspaces.json`;
-const urlAppServer = `http://${ipAddress}${serverPort}/api/worlds`;
+const urlGeoserverRest = `http://${config.ipAddress}${config.geoserverPort}/geoserver/rest/workspaces.json`;
```
Comment (yairtawil) [inline api/worlds.js]: config

File: index.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
 const express = require('express');
 const cors = require('cors');
 const api = require('./api');
-const config = require('./config/config');
+const { serverPort } = require('./config/configJson');
 
 const app = express();
```
Comment (yairtawil) [inline index.js:6]: 👍

File: old/index.js [hunk 7]
```diff
@@ -1,148 +0,0 @@
-const   formidable = require('formidable'),
```
Comment (yairtawil) [inline old/index.js:1]: wowwwww 👑

File: package.json [hunk 8]
```diff
@@ -13,13 +13,15 @@
   "license": "ISC",
   "dependencies": {
     "JSONStream": "^1.3.3",
+    "archiver": "^2.1.1",
```
Comment (yairtawil) [inline package.json]: `archiver` to dev

**Review by yairtawil — CHANGES_REQUESTED** [review]

File: api/uploadFiles.js (hunk 4 above)
Comment (dafnaweiss) [reply api/uploadFiles.js]: not yet. the zip ins't working OK - making an zip file that is not an archive. the comments are the 'archiver' package.

**Pull request merged**